pytorch_lightning/metrics/metric.py

```diff
@@ -74,15 +172,25 @@ def __init__(self, name: str,
                 Defaults to sum.
         """
         super().__init__(name)
-        self._orig_call = tensor_metric(group=reduce_group,
-                                        reduce_op=reduce_op)(super().__call__)
+        self.reduce_group = reduce_group
+        self.reduce_op = reduce_op
+
+    @staticmethod
+    def input_convert(self, data: Any):
+        return apply_to_collection(data,
+                                   (torch.Tensor, np.ndarray, numbers.Number),
+                                   convert_to_tensor,
+                                   self.dtype, self.device)
 
-    def __call__(self, *args, **kwargs) -> torch.Tensor:
-        def _to_device_dtype(x: torch.Tensor) -> torch.Tensor:
-            return x.to(device=self.device, dtype=self.dtype, non_blocking=True)
+    @staticmethod
+    def output_convert(self, data: Any, output: Any):
+        return apply_to_collection(output, torch.Tensor, convert_to_tensor,
```

Comment:
```suggestion
        return apply_to_collection(output, (torch.Tensor, np.ndarray, numbers.Number), convert_to_tensor,
```
shall it be the same as `TensorCollectionMetric`?

Reply:
`TensorMetric` assumes that everything is tensors whereas the `TensorMetricCollection` users can also output numpy arrays ect. In the future we should probably collaps these into one class (`TensorMetric` is restricted version of `TensorMetricCollection`).